test/src/test/java/org/locationtech/geowave/test/basic/GeoWaveEnumIndexIT.java

```diff
@@ -0,0 +1,231 @@
+      throws IOException {
+    final DataStore ds = dataStoreOptions.createDataStore();
+    final SimpleFeatureType type = initType();
+
+    ds.addType(
+        new FeatureDataAdapter(type),
+        createEnumIndex(
+            new String[] {"Eastern", "Central", "Mountain", "Pacific"},
+            type.indexOf(attrNameToIndex)));
+    if (includeSpatial) {
+      ds.addIndex(TYPE_NAME, DimensionalityType.SPATIAL.getDefaultIndices());
+    }
+    try (
+        InputStreamReader reader =
+            new InputStreamReader(
+                GeoWaveTextIndexIT.class.getClassLoader().getResourceAsStream(
+                    TEST_STATE_CAPITALS_RESOURCE_PATH));
+        Writer<SimpleFeature> w = ds.createWriter(TYPE_NAME)) {
+      Streams.stream(
+          CSVFormat.DEFAULT.withHeader(
+              "state",
+              "city",
+              "lon",
+              "lat",
+              "year",
+              "area",
+              "population",
+              "notes").parse(reader)).map(r -> toFeature(r, type)).forEach(w::write);
+    }
+  }
+
+  private static SimpleFeature toFeature(final CSVRecord r, final SimpleFeatureType t) {
+    final SimpleFeatureBuilder b = new SimpleFeatureBuilder(t);
+    final double longitude = Double.parseDouble(r.get("lon"));
+    b.add(
+        GeometryUtils.GEOMETRY_FACTORY.createPoint(
+            new Coordinate(longitude, Double.parseDouble(r.get("lat")))));
+    b.add(r.get("state"));
+    b.add(r.get("city"));
+    b.add(Integer.parseInt(r.get("year")));
+    b.add(Double.parseDouble(r.get("area")));
+    b.add(Integer.parseInt(r.get("population")));
+    b.add(r.get("notes"));
+    b.add(getTimeZone(longitude));
+    return b.buildFeature(UUID.randomUUID().toString());
+  }
+
+  private static String getTimeZone(final double longitude) {
+    if (longitude < -117) {
+      return "Pacific";
+    } else if (longitude < -101) {
+      return "Mountain";
+    } else if (longitude < -86.2) {
+      return "Central";
+    }
+    return "Eastern";
+  }
+
+  private static Index createEnumIndex(final String[] enumValues, final int attributeIndex) {
+    return new CustomIndex<>(
+        new EnumIndexStrategy<>(new VectorTextIndexEntryConverter(attributeIndex), enumValues),
+        TEST_ENUM_INDEX_NAME);
+  }
+
+  @Test
+  public void testEnumIndex() throws IOException {
+    assertResults(false);
+  }
+
+  @Test
+  public void testEnumIndexWithSpatial() throws IOException {
+    assertResults(true);
+  }
+
+  private void assertResults(final boolean includeSpatial) throws IOException {
+    ingest(TIMEZONE_ATTR_NAME, includeSpatial);
+
+    getResults(new EnumSearch("Pacific"));
+    getResults(new EnumSearch("Mountain"));
+    getResults(new EnumSearch("Central"));
+    getResults(new EnumSearch("Eastern"));
+    // intentionally make sure a bogus term just returns no results
+    getResults(new EnumSearch("Atlantic"));
+  }
+
+  private void getResults(final EnumSearch search) {
```

Comment:
Wouldn't both of these queries use the same index because of the custom constraints?  Seems like if for some reason some entries were being left out of the result it wouldn't be caught.  Maybe there should be an assertion for the number of results in each set as well.

Reply:
good catch!  I intended to leave the first query collecting expected results as unconstrained and just apply a manual filter on the full dataset, but I guess I goofed.